### PR TITLE
Function.js was created for part 2 of javascript

### PR DESCRIPTION
**Pull request by karlysun**

### CODE REVIEW

File: functions.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
+// This file was created for javascript part 2 class
+
+// First exercice: Create an array prototype to sort two strings by longest
+
+function sortLongestString(a,b){
```
Comment (ziad-saab) [inline functions.js]: :+1:

File: functions.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    return b.length - a.length;
+}
+
+var yourArray = ["Allo","Bonjour"];
+
+yourArray.sort(sortLongestString);
+
+// Function that will sort array by longest name, then longest e-mail
+
+function sortLongestName(a,b){
```
Comment (ziad-saab) [inline functions.js]: :+1: but you can simply write `b.name.length`. The parentheses are not needed here :)

File: functions.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    
+    return (b.name).length - (a.name).length ;
+}
+
+function sortAlphaEmail(a,b){
+    
+    return (a.email).localeCompare(b.email) ;
```
Comment (ziad-saab) [inline functions.js]: Nice! Tell me a bit more about localeCompare please?

File: functions.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+                    {name: "Pat", email: "asdf_as@hotmail.com"},
+                    {name: "Thomas", email: "zsdf__34@gmail.com"},
+                    {name: "Johnny-O", email: "xsdf.askd@hotmail666.com"} ];
+
+console.log(yourArray.sort(sortLongestName));
+console.log(yourArray2.sort(sortAlphaEmail));
+
+// Create a function that will return the square of a number
+
+function squareOfNumber(value){
```
Comment (ziad-saab) [inline functions.js]: :+1:

File: functions.js [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+var yourArray = [{num: 12},{num: 20},{num: 4},{num: 102}];
+
+yourArray.map(squareOfNum);
+
+// Create a function that will take an operation and two numbers and will return function
+
+function operationMaker(operation){
+    
+    function doingTheMaths(a,b){
```
Comment (ziad-saab) [inline functions.js]: This is great. For an additional challenge, can you please put the `switch` outside the function, and return one of four functions that does only the requested operation?

File: functions.js (hunk 3 above)
Comment (karlysun) [inline functions.js]: The method compares if the `a.email` goes before or after (alpahabeticaly) the `b.email`

**Pull request closed without merging**